lib/myxql/protocol/types.ex

```diff
@@ -66,6 +66,8 @@ defmodule MyXQL.Protocol.Types do
     string
   end
 
+  def take_string_nul(""), do: {nil, ""}
```

Comment:
This was what caused crashes when spinning it up in a Phoenix app, `mysql_clear_password` doesn't have any scramble data so it's just empty.